presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -0,0 +1,455 @@
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        checkArgument(!DEFAULT_SCHEMA.equalsIgnoreCase(schemaName), "Can't delete 'default' schema which maps to Phoenix empty schema");
+        phoenixClient.execute(session, format("DROP SCHEMA %s", getEscapedArgument(toMetadataCasing(session, schemaName))));
+    }
+
+    private String toMetadataCasing(ConnectorSession session, String schemaName)
+    {
+        try (Connection connection = phoenixClient.getConnection(JdbcIdentity.from(session))) {
+            boolean uppercase = connection.getMetaData().storesUpperCaseIdentifiers();
+            if (uppercase) {
+                schemaName = schemaName.toUpperCase(ENGLISH);
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(PHOENIX_METADATA_ERROR, "Couldn't get casing for the schema name", e);
+        }
+        return schemaName;
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+    {
+        JdbcTableHandle existingTable = getTableHandle(session, tableMetadata.getTable());
+        if (existingTable != null && !ignoreExisting) {
+            throw new IllegalArgumentException("Target table already exists: " + tableMetadata.getTable());
+        }
+        createTable(session, tableMetadata);
+    }
+
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorNewTableLayout> layout)
+    {
+        JdbcOutputTableHandle handle = createTable(session, tableMetadata);
+        return handle;
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession session, ConnectorOutputTableHandle tableHandle, Collection<Slice> fragments, Collection<ComputedStatistics> computedStatistics)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        JdbcTableHandle handle = (JdbcTableHandle) tableHandle;
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, tableHandle, true);
+        List<ColumnMetadata> nonRowkeyCols = tableMetadata.getColumns().stream().filter(column -> !ROWKEY.equalsIgnoreCase(column.getName())).collect(Collectors.toList());
+        return new PhoenixOutputTableHandle(
+                Optional.ofNullable(handle.getCatalogName()),
+                Optional.ofNullable(handle.getSchemaName()),
+                handle.getTableName(),
+                nonRowkeyCols.stream().map(ColumnMetadata::getName).collect(Collectors.toList()),
+                nonRowkeyCols.stream().map(ColumnMetadata::getType).collect(Collectors.toList()),
+                "", // no use of temporary tables
```

Comment:
How does this work? This field is used in several places in `BaseJdbcClient` which are not overridden here, and there is no special casing for empty.

If it is somehow never used, it would be cleaner to remove it from the constructor of `PhoenixOutputTableHandle` and pass `""` when calling `super()`. This is fine since Jackson will ignore properties that don't have a constructor parameter when deserializing.